### PR TITLE
FIX-#6587: Use different env files for unidist engine for windows and linux

### LINKED ISSUE #6587: Use different env files for unidist engine for windows and linux
unidist depends on mpi4py only and conda solver may pick different MPI implementations in depend on a concrete development envionment. We should be explicit and specify the exact MPI implementation we test on.

### PR DESCRIPTION
**Pull request by YarShev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6587  
- [x] tests passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -556,7 +556,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: ./.github/actions/mamba-env
         with:
-          environment-file: ${{ matrix.execution.name == 'unidist' && 'requirements/env_unidist.yml' || 'environment-dev.yml' }}
+          environment-file: ${{ matrix.os = 'ubuntu' && matrix.execution.name == 'unidist' && 'requirements/env_unidist_linux.yml' || matrix.os = 'windows' && matrix.execution.name == 'unidist' && 'requirements/env_unidist_win.yml' || 'environment-dev.yml' }}
```
Comment (YarShev) [inline .github/workflows/ci.yml:559]:
```suggestion
          environment-file: ${{ matrix.os == 'ubuntu' && matrix.execution.name == 'unidist' && 'requirements/env_unidist_linux.yml' || matrix.os == 'windows' && matrix.execution.name == 'unidist' && 'requirements/env_unidist_win.yml' || 'environment-dev.yml' }}
```

File: requirements/env_unidist_linux.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ dependencies:
   - pandas>=2.1,<2.2
   - numpy>=1.22.4
   - unidist-mpi>=0.2.1
+  - mpich
```
Comment (vnlitvinov) [inline requirements/env_unidist_linux.yml:11]: I wonder why it is not a Linux-specific dependency of `unidist-mpi` instead...

Comment (YarShev) [reply requirements/env_unidist_linux.yml:11]: If we added any MPI library as a linux-specific dependency of `unidist-mpi`, we would jeopardize the option of using a system-provided MPI library.

**Review by anmyachev — APPROVED** [review]
LGTM!

**Pull request merged**